monero-harness/src/lib.rs

```diff
@@ -106,17 +106,34 @@ impl<'c> Monero {
         tracing::info!("Generated {:?} blocks", res.blocks.len());
         miner_wallet.refresh().await?;
 
-        for (wallet, amount) in wallet_amount.iter() {
-            if *amount > 0 {
-                let wallet = self.wallet(wallet)?;
-                let address = wallet.address().await?.address;
-                miner_wallet.transfer(&address, *amount).await?;
+        Ok(())
+    }
+
+    pub async fn init_wallet(&self, name: &str, amount_in_outputs: Vec<u64>) -> Result<()> {
+        let miner_wallet = self.wallet("miner")?;
+        let miner_address = miner_wallet.address().await?.address;
+        let monerod = &self.monerod;
+
+        let wallet = self.wallet(name)?;
+        let address = wallet.address().await?.address;
+
+        for amount in amount_in_outputs {
+            if amount > 0 {
+                miner_wallet.transfer(&address, amount).await?;
                 tracing::info!("Funded {} wallet with {}", wallet.name, amount);
                 monerod.client().generate_blocks(10, &miner_address).await?;
```

Comment:
Note: I tried "optimizing" this by doing all the transfers and then generate 10 blocks (i.e. generate the blocks after the loop), but it had weird side effects on the complete test setup where Balances where suddenly not properly unlocked for assertion.
I am not completely sure why it would behave like that, but for no I would keep it like this.

Reply:
I had this optimization in a separate commit and dropped it :)